### PR TITLE
cargo doc generation dispatch workflow

### PR DESCRIPTION
**Pull request by zees-dev**

### CODE REVIEW

**Review by JasonTulp — APPROVED** [review]
Damn nice, this will be useful :+1:

File: .github/workflows/generate-docs.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        ref: ${{ github.event.inputs.branch }}
+        fetch-depth: 0
+
+    - uses: dtolnay/rust-toolchain@1.69
+
+    - name: Install toolchain
+      run: rustup show
+
+    - name: Generate Documentation
+      run: cargo doc -p ${{ github.event.inputs.pallet_name }}
```
Comment (surangap) [inline .github/workflows/generate-docs.yml:37]: might need to correct the input name. also might have to think about having multiple pallet names at once?

**Review by surangap — COMMENTED** [review]
would be useful.

File: .github/workflows/generate-docs.yml (hunk 1 above)
Comment (zees-dev) [reply .github/workflows/generate-docs.yml:37]: will iterate on it once it's merged; just wanted quick win for automating this

**Pull request merged**